src/mbi/relaxed_projection.py

```diff
@@ -0,0 +1,368 @@
+import numpy as np
+import pandas as pd
+import itertools
+from typing import Callable, List, Tuple
+
+import jax
+import jax.numpy as jnp
+import optax
+from .dataset import Dataset
+from .domain import Domain
+
+
+
+#######################################################
+#                                                     #
+#       Here is a the projectable data class          #
+#                                                     #
+#######################################################
+
+class ProjectableData():
+    def __init__(
+            self, 
+            D: jax.Array, 
+            domain: Domain
+        ):
+        """
+        A class storing one-hot data, supporting projection, vector 
+        query, and synthesize data operation.
+        
+        Args:
+            D: a jax.Array, representing a one-hot form of the data.
```

Comment:
I'm not sure one_hot is the right term here, since the values in D need not be 0 or 1.  The term "soft_hot" would be a bit better here.  Once you decide on a good name, change D accordingly (e.g., "D" -> "soft_hot_data")